components/diagnostics/diagnose_me/component.py

```diff
@@ -29,134 +33,192 @@ def run_diagnose_me(
               If set to HALT_ON_ERROR will case any error to raise an exception.
               This is intended to stop the data processing of a pipeline. Can set
               to False to only report Errors/Warnings.
+          project_id:
+              GCP project ID which is assumed to be accessible from the pod.
           target_apis:
               String consisting of a comma separated list of apis to be verified.
+          quota_check:
+              List of entries describing how much quota is required. Each entry
+              has three fields: region, metric and quota_needed. All
+              string-typed.
       Raises:
           RuntimeError: If configuration is not setup properly and
           HALT_ON_ERROR flag is set.
       """
 
-    # Installing pip3 and kfp, since the base image 'google/cloud-sdk:279.0.0'
-    # does not come with pip3 pre-installed.
-    import subprocess
-    subprocess.run(
-        ['curl', 'https://bootstrap.pypa.io/get-pip.py', '-o', 'get-pip.py'],
-        capture_output=True)
-    subprocess.run(['apt-get', 'install', 'python3-distutils', '--yes'],
-                   capture_output=True)
-    subprocess.run(['python3', 'get-pip.py'], capture_output=True)
-    subprocess.run(['python3', '-m', 'pip', 'install', 'kfp>=0.1.31', '--quiet'],
-                   capture_output=True)
-
-    import sys
-    from typing import List, Text
-    import os
-    from kfp.cli.diagnose_me import gcp
-    from kfp.cli.diagnose_me import utility
-    import json as json_library
-
-    config_error_observed = False
-
-    # Get the project ID
-    # from project configuration
-    project_config = gcp.get_gcp_configuration(
-        gcp.Commands.GET_GCLOUD_DEFAULT, human_readable=False)
-    if not project_config.has_error:
-        auth_project_id = project_config.parsed_output['core']['project']
-        print('GCP credentials are configured with access to project: %s ...\n' % (project_id))
-        print('Following account(s) are active under this pipeline:\n')
-        subprocess.run(['gcloud', 'auth', 'list','--format','json'])
-        print('\n')
+  # Installing pip3 and kfp, since the base image 'google/cloud-sdk:279.0.0'
+  # does not come with pip3 pre-installed.
+  import subprocess
+  subprocess.run([
+      'curl', 'https://bootstrap.pypa.io/get-pip.py', '-o', 'get-pip.py'
+  ],
+                 capture_output=True)
+  subprocess.run(['apt-get', 'install', 'python3-distutils', '--yes'],
+                 capture_output=True)
+  subprocess.run(['python3', 'get-pip.py'], capture_output=True)
+  subprocess.run(['python3', '-m', 'pip', 'install', 'kfp>=0.1.31', '--quiet'],
+                 capture_output=True)
+
+  import sys
+  from kfp.cli.diagnose_me import gcp
+
+  config_error_observed = False
+
+  quota_list = gcp.get_gcp_configuration(
+      gcp.Commands.GET_QUOTAS, human_readable=False
+  )
+  quota_dict = {}  # Mapping from region to dict[metric, available]
```

Comment:
May want to add a logic to handle the case were `gcp.get_gcp_configuration` errors out. 

```
if not quota_list.has_error:
  for region_quota ...
else:
 print(
        'Failed to retrieve Project quota with error  %s\n' %
        (quota_list.stderr),
        file=sys.stderr
    )
    config_error_observed = True
```

Reply:
Nice catch. Thanks!